cli/command/container/completion.go

```diff
@@ -158,19 +158,24 @@ func completeDetachKeys(_ *cobra.Command, _ []string, _ string) ([]string, cobra
 // to provide completion for CDI devices that were discovered by the daemon.
 func completeCDIDevices(dockerCLI completion.APIClientProvider) cobra.CompletionFunc {
 	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		if strings.HasPrefix(toComplete, "/") {
+			return nil, cobra.ShellCompDirectiveDefault | cobra.ShellCompDirectiveNoSpace
+		}
+
 		info, err := dockerCLI.Client().Info(cmd.Context())
```

Comment:
Opening as draft, because I had this branch but after implementing CDI completion, I realised we also need to provide completion for file-paths (assuming the daemon and CLI are on the same host), so we need some solid way to distinguish "user wants to complete a CDI device" vs "user wants to add a _device_ (e.g. `/dev/foo:/some/path`).